### DISCUSSION TITLE
How to install the latest version on Ubuntu/Mint?

### DISCUSSION THREAD

**Discussion by dusanbrankov** [i3/i3#5835] [Q&A] [answered]
I just can't figure out how to install the latest stable version of i3 on Linux Mint. I first installed it via `apt` and got version `4.20.1`, which doesn't seem to contain `i3-gaps`.

Now I have downloaded the current version `i3-4.23.tar.xz` from the official website, but there is no `config` file, so how do I install it?

Usually I'm running

```sh
./config
make
sudo make install
```

Searching the internet didn't bring me any further either.

Thanks.

P.S. I'm also not able to install `i3-gaps` separately because of lack of `jammy` support...

**Comment by okraits** [comment]
You can also have a look at https://i3wm.org/docs/repositories.html

**Reply by dusanbrankov** [reply]
Thanks for the reply. I already tried that, it doesnt install or update anything, it just says

```
i3 is already the newest version (4.20.1-1)
```

I think the most straightforward way should be to install from source, but I can't find any instruction on how to install without a `configure` executable.

**Reply by okraits** [reply]
You can find instructions on how to build i3 from source in the following places:

https://i3wm.org/docs/hacking-howto.html#_building_i3
https://github.com/Airblader/i3/wiki/Building-from-source

**Reply by dusanbrankov** [reply]
Ok, I installed Mason and ran:

```sh
mkdir build && meson setup $_
```

But I got the following error:

```
meson.build:305:0: ERROR: Dependency "libstartup-notification-1.0" not found, tried pkgconfig
```

There is no such package on `apt`, only `libstartup-notification0` which I have already installed.

I don't understand why they make things so fucking complicated. All I want is some pixels of gap around my windows, but I will leave it as is and search for an alternative.

Thanks anyway, appreciate your help.

**Reply by orestisfl** [reply, marked as answer]
Please prefer to use your package manager instead of building software from source. I've never used Mint but I found this https://forums.linuxmint.com/viewtopic.php?t=388673:

> I followed the instructions on https://i3wm.org/docs/repositories.html for using i3's Ubuntu repositories.
> 
> I edited the generated /etc/apt/sources.list.d/sur5r-i3.list to change 'vera' to 'jammy'.
> 
> From there all went fine, did sudo apt upgrade to bump i3 4.20 to 4.22.

**Reply by dusanbrankov** [reply]
@orestisfl It works, thank you very much.

**Reply by lovestaco** [reply]
Hi how did you get the 4.23 installed?
I updated using package manager and it still says 4.22.

Not able to add gaps

**Reply by dusanbrankov** [reply]
@lovestaco I actually upgraded from 4.20 to 4.22, but you can follow the same steps as I did to upgrade to the current version, following the instructions at https://i3wm.org/docs/repositories.html, as already mentioned by @orestisfl.

```sh
$ /usr/lib/apt/apt-helper download-file https://debian.sur5r.net/i3/pool/main/s/sur5r-keyring/sur5r-keyring_2024.03.04_all.deb keyring.deb SHA256:f9bb4340b5ce0ded29b7e014ee9ce788006e9bbfe31e96c09b2118ab91fca734
$ sudo apt install ./keyring.deb
$ echo "deb http://debian.sur5r.net/i3/ $(grep '^DISTRIB_CODENAME=' /etc/lsb-release | cut -f2 -d=) universe" | sudo tee /etc/apt/sources.list.d/sur5r-i3.list
$ sudo apt update
```

You should then be able to upgrade i3 to the latest version (I just did the same):

```sh
$ apt list --upgradable
i3-wm/unknown 4.23-1~~jammy1 amd64 [upgradable from: 4.22-2~~jammy1]
i3/unknown 4.23-1~~jammy1 amd64 [upgradable from: 4.22-2~~jammy1]

$ sudo apt upgrade
$ i3 -v
i3 version 4.23 (2023-10-29) © 2009 Michael Stapelberg and contributors
```
